dhis-2/dhis-api/src/main/java/org/hisp/dhis/period/Period.java

```diff
@@ -75,7 +74,7 @@ public static boolean isDateInTimeFrame(
     }
     ZoneId zoneId = ZoneId.systemDefault();
     Function<Date, ZonedDateTime> toZonedDate =
-        date -> ZonedDateTime.ofInstant(date.toInstant(), zoneId).with(LocalTime.MIN);
```

Comment:
Have you checked that all callers now do want more than full day resolution? I think I added this method some time back to evaluate period related logic and periods are always full days.

Reply:
@jbee I have added another method which mirrors this one, but considers exact dates with times instead.